ttnn/cpp/ttnn/operations/moreh/moreh_dot_op_backward/device/moreh_dot_backward_program_factory.cpp

```diff
@@ -0,0 +1,226 @@
+                auto *dst0_buffer = input_grad_tensor.buffer();
+                dst0_address = dst0_buffer->address();
+            }
+
+            if (has_other_grad) {
+                const auto &other_grad_tensor = other_grad.value();
+                auto *dst1_buffer = other_grad_tensor.buffer();
+                dst1_address = dst1_buffer->address();
+            }
+
+            CoreCoord core = {0, 0};
+            {
+                auto &runtime_args = GetRuntimeArgs(program, reader_kernel_id, core);
+                runtime_args[0] = (std::uint32_t)has_input_grad;
+                runtime_args[1] = (std::uint32_t)has_input_grad;
+                runtime_args[2] = src0_buffer->address();
+                runtime_args[3] = src1_buffer->address();
+                runtime_args[4] = src2_buffer->address();
+            }
+
+            {
+                auto &runtime_args = GetRuntimeArgs(program, compute_kernel_id, core);
+                runtime_args[0] = (std::uint32_t)has_input_grad;
+                runtime_args[1] = (std::uint32_t)has_input_grad;
+            }
+
+            {
+                auto &runtime_args = GetRuntimeArgs(program, writer_kernel_id, core);
+                runtime_args[0] = (std::uint32_t)has_input_grad;
+                runtime_args[1] = (std::uint32_t)has_input_grad;
+                runtime_args[2] = dst0_address;
+                runtime_args[3] = dst1_address;
+            }
+        };
+    return {
+        std::move(program),
+        {.unary_reader_kernel_id = reader_kernel_id, .unary_writer_kernel_id = writer_kernel_id}};
+}
+
+void MorehDotBackwardOperation::SingleCore::override_runtime_arguments(
+    cached_program_t& cached_program,
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& tensor_return_value) {
+    auto& program = cached_program.program;
+    auto& unary_reader_kernel_id = cached_program.shared_variables.unary_reader_kernel_id;
+    auto& unary_writer_kernel_id = cached_program.shared_variables.unary_writer_kernel_id;
+
+    const auto& output_grad_buffer = tensor_args.output_grad.buffer();
+    const auto& input_buffer = tensor_args.input.buffer();
+    const auto& other_buffer = tensor_args.other.buffer();
+    const auto& input_grad_buffer = tensor_return_value.at(0).buffer();
+    const auto& other_grad_buffer = tensor_return_value.at(1).buffer();
+
+    {
+        auto& runtime_args = tt::tt_metal::GetRuntimeArgs(program, unary_reader_kernel_id, CoreCoord{0, 0});
+        runtime_args[0] = output_grad_buffer->address();
+        runtime_args[1] = input_buffer->address();
+        runtime_args[2] = other_buffer->address();
+    }
+
+    {
+        auto& runtime_args = tt::tt_metal::GetRuntimeArgs(program, unary_writer_kernel_id, CoreCoord{0, 0});
+        runtime_args[0] = input_grad_buffer->address();
+        runtime_args[1] = other_grad_buffer->address();
```

Comment:
```suggestion
        runtime_args[2] = input_grad_buffer->address();
        runtime_args[3] = other_grad_buffer->address();
```
In the write kernel, the address is the 2nd and 3rd runtime arguments.

https://github.com/tenstorrent/tt-metal/blob/c09e014b6f486680ca89e442cb2724e84466cc4c/ttnn/cpp/ttnn/operations/moreh/moreh_dot_op_backward/device/kernels/writer_moreh_dot_backward.cpp#L10